src/hooks/write/useCreateTournament.ts

```diff
@@ -40,26 +38,49 @@ export function useCreateTournament(
       }
     : undefined;
 
-  const { config, error: err } = usePrepareContractWrite({
+  const {
+    config,
+    error: prepareContractWriteError,
+    isError: isPrepareContractWriteError,
+  } = usePrepareContractWrite({
     addressOrName: chainData.defifaCreate.address,
     contractInterface: chainData.defifaCreate.interface,
     functionName: "launchGameWith",
     args: [preparedLaunchProjectData],
     chainId: chainData.chainId,
   });
+  if (isPrepareContractWriteError) {
+    console.error(
+      "useCreateTournament::usePrepareContractWriteError::error",
+      prepareContractWriteError
+    );
+  }
 
   const { data, write, error, isError } = useContractWrite(config);
-  
-  console.log("payload", preparedLaunchProjectData);
-  console.log("Contract call:", config);
-  console.log("Contract call data:", data);
+  if (isError) {
+    console.error("useCreateTournament::useContractWrite::error", error);
+  }
 
-  const { isLoading, isSuccess } = useWaitForTransaction({ hash: data?.hash });
+  const {
+    isLoading,
+    isSuccess,
+    isError: isWaitForTransactionError,
+    error: waitForTransactionError,
+  } = useWaitForTransaction({ hash: data?.hash });
+  if (isWaitForTransactionError) {
+    console.error(
+      "useCreateTournament::useWaitForTransaction::error",
+      waitForTransactionError
+    );
+  }
 
   const handleWrite = () => {
     if (!isConnected) {
       openConnectModal!();
     } else {
+      console.log("useCreateTournament::payload", preparedLaunchProjectData);
+      console.log("useCreateTournament::Contract call:", config);
+      console.log("useCreateTournament::Contract call data:", data);
```

Comment:
only log this stuff when someone clicks the button - i think this is the only time that these logs are useful!